python_modules/dagster/dagster/_core/definitions/asset_check_spec.py

```diff
@@ -18,13 +18,13 @@
 @experimental
 @whitelist_for_serdes
 class AssetCheckSeverity(Enum):
-    """Severity level for an asset check.
+    """Severity level for an asset check. Severity determines how the check result is displayed in
+    the UI.
 
     Severities:
 
-    - WARN: If the check fails, don't fail the step.
-    - ERROR: If the check fails, fail the step and, within the run, skip materialization of any
-      assets that are downstream of the asset being checked.
+    - ERROR: If the check fails, the asset will be colored red in the UI.
+    - WARN: Does not affect the color of the asset in the UI.
```

Comment:
I don't love this description. We usually don't mention the UI in api docs afaik and it could get out of sync. This is the only hard difference between error and warn though

Reply:
> We usually don't mention the UI in api docs afaik and it could get out of sync

Yeah, I share this concern. I would lean away from framing severity in terms of how it shows up in the UI. In addition to the concern about the UI getting out of sync, there are also other places where people consume severities outside of our asset UI - e.g. in alerts. And they can be accessed using the Python API. I would lean towards something along the lines of this: https://betterstack.com/community/guides/logging/log-levels-explained/.

I.e. "WARN" indicates that something unexpected has occurred, "ERROR" indicates that something clearly problematic has occurred.

I also think it's worth calling out explicitly that neither of them affect control flow.

Reply:
updated to potential vs definite issue with the asset